File: 通用代码/命令模式(Reciver封装)/ConcreteCommand1.java
public class ConcreteCommand1 extends Command{
    //声明自己的默认接收者
    public ConcreteCommand1(){
        super(new ConcreteReciver1());
    }
    //构造函数传递接收者
    public ConcreteCommand1(Receiver _receiver){
        super(_receiver);
    }
    //必须实现一个命令
    public void execute(){
        //业务处理
        this.receiver.doSomething();
    }
}

File: 通用代码/命令模式(Reciver封装)(C++)/ConcreteCommand1.cpp
#include "ConcreteCommand1.h"

ConcreteCommand1::ConcreteCommand1(void)
{
    //声明自己的默认接收者
    this->Command::_pReciever = new ConcreteReceiver1();
    cout << "ConcreteCommand1..." << endl;
}

ConcreteCommand1::ConcreteCommand1(Receiver* pReceiver)
{
    this->Command::_pReciever = pReceiver;
    cout << "ConcreteCommand1..." << endl;
}

ConcreteCommand1::~ConcreteCommand1()
{
    cout << "~ConcreteCommand1..." << endl;
}

void ConcreteCommand1::Execute()
{
    this->Command::_pReciever->doSomething();
}

